### PR TITLE
Add traceparent to UCloudEventAttributes

### PR DESCRIPTION
**Pull request by jefftindall**
To support the [proposed update](https://github.com/eclipse-uprotocol/uprotocol-spec/pull/48) to the uProtocol-spec with distributed tracing support.

### CODE REVIEW

File: src/main/java/org/eclipse/uprotocol/cloudevent/datamodel/UCloudEventAttributes.java [hunk 1]
```diff
@@ -40,28 +40,32 @@ public class UCloudEventAttributes {
...
-     * @param hash     an HMAC generated on the data portion of the CloudEvent message using the device key.
+     * @param hash     An HMAC generated on the data portion of the CloudEvent message using the device key.
      * @param priority uProtocol Prioritization classifications.
      * @param ttl      How long this event should live for after it was generated (in milliseconds).
      *                 Events without this attribute (or value is 0) MUST NOT timeout.
      * @param token    Oauth2 access token to perform the access request defined in the request message.
+     * @param traceparent    Optional identifier used to correlate observability across related events
      */
-    private UCloudEventAttributes(String hash, UPriority priority, Integer ttl, String token) {
+    private UCloudEventAttributes(String hash, UPriority priority, Integer ttl, String token, String traceparent) {
```
Comment (stevenhartley) [inline src/main/java/org/eclipse/uprotocol/cloudevent/datamodel/UCloudEventAttributes.java:55]: Optional attributes are not passed to the constructor, I fully expect traceparent to be an optional attribute

**Review by stevenhartley — CHANGES_REQUESTED** [review]
See comments and we need to add an issue to the spec and link to this project and all the other SDKs. I'm not against this infact others have asked for it but we need to make sure we get this right

File: src/main/java/org/eclipse/uprotocol/cloudevent/datamodel/UCloudEventAttributes.java (hunk 1 above)
Comment (jefftindall) [reply src/main/java/org/eclipse/uprotocol/cloudevent/datamodel/UCloudEventAttributes.java:68]: Thanks for the feedback @stevenhartley .  I overloaded the constructor and it is now optional.  Does this resolve your comment, or are you suggesting that optional parameters may only be set through the builder pattern.

Comment (stevenhartley) [reply src/main/java/org/eclipse/uprotocol/cloudevent/datamodel/UCloudEventAttributes.java:68]: That is not what I meant :-). constructor is for mandatory parameters, don't touch the constructor and add a builder method for this parameter.

**Review by stevenhartley — CHANGES_REQUESTED** [review]

File: src/main/java/org/eclipse/uprotocol/cloudevent/datamodel/UCloudEventAttributes.java (hunk 1 above)
Comment (jefftindall) [reply src/main/java/org/eclipse/uprotocol/cloudevent/datamodel/UCloudEventAttributes.java]: Done

**Pull request merged**